portfolio/src/main/java/com/google/sps/servlets/DataServlet.java

```diff
@@ -76,17 +107,39 @@ public void doPost(HttpServletRequest request, HttpServletResponse response) thr
 
     String argURL = request.getParameter("url-input");
     String argComment = request.getParameter("comment-input");
+
+    if(
+        !(
+            ( Pattern.matches("^http:\\/\\/(.+)\\.sinaimg\\.cn\\/large\\/(.+)\\.(jpg|png|gif)$",argURL) )
```

Comment:
Don't need to escape slash "/" in a regular expression.

I have tried to upload a meme using one of the provided sina URL, but failed. Maybe the escape is the root cause. Try fixing the pattern?

![image](https://user-images.githubusercontent.com/635513/87412251-64649d00-c5fb-11ea-8c74-b95b9ee4818c.png)